src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_adapter.py

```diff
@@ -0,0 +1,750 @@
+from ...schedulers import KarrasDiffusionSchedulers
+from ...utils import (
+    PIL_INTERPOLATION,
+    is_accelerate_available,
+    is_accelerate_version,
+    logging,
+    randn_tensor,
+    replace_example_docstring,
+)
+from ..pipeline_utils import DiffusionPipeline
+from . import StableDiffusionPipelineOutput
+from .safety_checker import StableDiffusionSafetyChecker
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> from PIL import Image
+        >>> from diffusers.utils import load_image
+
+        >>> image = load_image("https://huggingface.co/RzZ/sd-v1-4-adapter-color/resolve/main/color_ref.png")
+
+        >>> color_palette = image.resize((8, 8))
+        >>> color_palette = color_palette.resize((512, 512), resample=Image.Resampling.NEAREST)
+
+        >>> import torch
+        >>> from diffusers import StableDiffusionAdapterPipeline, Adapter
+
+        >>> adapter = Adapter.from_pretrained("RzZ/sd-v1-4-adapter-color")
+        >>> pipe = StableDiffusionAdapterPipeline.from_pretrained(
+        ...     "CompVis/stable-diffusion-v1-4",
+        ...     adapter=adapter,
+        ...     torch_dtype=torch.float16,
+        ... )
+
+        >>> pipe.to("cuda")
+
+        >>> out_image = pipe(
+        ...     "At night, glowing cubes in front of the beach",
+        ...     image=color_palette,
+        ...     generator=generator,
+        ... ).images[0]
+        ```
+"""
+
+
+def preprocess(image, target_height, target_width):
+    if isinstance(image, torch.Tensor):
+        return image
+    elif isinstance(image, PIL_Image.Image):
+        image = [image]
+
+    multi_control = isinstance(image[0], list) or isinstance(image[0], tuple)
+    multi_control |= isinstance(image[0], torch.Tensor) and image[0].ndim == 4
+
+    if multi_control:
+        images = [preprocess(subset, target_height, target_width) for subset in image]
+        b, c, h, w = images[0].shape
+        image = [img.reshape([b * c, h, w]) for img in images]
+
+    if isinstance(image[0], PIL_Image.Image):
+        image = [
+            np.array(i.resize((target_width, target_height), resample=PIL_INTERPOLATION["lanczos"]))[None, :]
+            for i in image
+        ]
+        image = np.concatenate(image, axis=0)
+        image = np.array(image).astype(np.float32) / 255.0
+        image = image.transpose(0, 3, 1, 2)
+        image = torch.from_numpy(image)
+    elif isinstance(image[0], torch.Tensor):
+        if image[0].ndim == 3:
+            size_align_4d = []
+            for img_ten in image:
+                h, w = img_ten.shape[-2:]
+                img_ten = torch.unsqueeze(img_ten, dim=0)
+                if (h, w) != (target_height, target_width):
+                    img_ten = torch.nn.functional.interpolate(img_ten, (target_height, target_width))
+                size_align_4d.append(img_ten)
+            image = torch.cat(size_align_4d, dim=0)
```

Comment:
by convention, we do not re-size tensor inputs. tensor inputs are handed as is to the models. If there is a dimension mismatch, that is ok :)

Reply:
Ah right, "tensor inputs are handed as is" as the doc string says.